dnase/callsnpsMerge.sh

```diff
@@ -163,6 +164,41 @@ bcftools index ${sampleOutdir}/${name}.${mappedgenome}.filtered.vcf.gz
 tabix -p vcf ${sampleOutdir}/${name}.${mappedgenome}.filtered.vcf.gz
 rm -f ${fltvcffiles}
 
+echo
+echo "Run Delly"
+date
+
+## Make a place for the temp bam files with qcfail turned off.
+bam_output_qcOK=${TMPDIR}/${name}.${mappedgenome}.QC_OK_bamfile.bam
+
+## Turn off all qcfail flags.
+python ${src}/changeFlags.py ${sampleOutdir}/${name}.${mappedgenome}.bam ${bam_output_qcOK}
+
+samtools index ${bam_output_qcOK}
+
+set +e
+## Look for variants (DEL INS DUP INV TRA).
+delly call -t ALL \
+    -o ${sampleOutdir}/${name}.${mappedgenome}.delly.bcf \
+    -g ${referencefasta} \
+    ${bam_output_qcOK}
+
+# If delly exited with a 0 code then create the vcf
+if [ $? -eq 0 ]; then
+  bcftools index ${sampleOutdir}/${name}.${mappedgenome}.delly.bcf
+  ## Create a vcf file, and set it up for viewing in vcfTabix format.
+  ## Only accept variants that pass the FILTER test.
+  bcftools filter -i 'FILTER="PASS" & PE>10' ${sampleOutdir}/${name}.${mappedgenome}.delly.bcf > ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf
+
+  bgzip -f ${sampleOutdir}/${name}.${mappedgenome}.delly.filtered.vcf
```

Comment:
can this be done using a pipe?

Reply:
Is there a reason for not using the `--output-type z` flag (compressed VCF)? This would merge the second command into the first.
There are other cases in the code where this could be done such as:
```sh
bcftools concat --output-type v ${fltvcffiles} | bgzip -c -@ $NSLOTS > ${sampleOutdir}/${name}.${mappedgenome}.filtered.vcf.gz;;
```

Reply:
double check me but I believe that will output gzip compressed rather than bgzip, which is needed for some block-indexed access applications

Reply:
I tried looking through their code, but was not successful.
This stackoverflow seems to indicate that it uses bgzip:
https://bioinformatics.stackexchange.com/questions/2890/bcftools-output-vs-bgzip

Reply:
Good catch.

However, changing to --output-type z would be done separately and regression tested. Why don't you make an issue and we can revisit it when we have time?

Reply:
That works, I will do it using the current format for now.

Reply:
Opened #263